src/types/json.ts

```diff
@@ -12,6 +12,11 @@ export const types = {
 export const fetchProcessing = (data: any) => JSON.parse(data);
 
 export const validate = TypeUtils.validate.checkRequired((value) => {
+	// Disallow primitives
+	if (typeof value !== 'object') {
```

Comment:
I guess we could also add `!(value instance of Date)` if we wanted to be super strict.

Reply:
Good catch, this would also be true for things like `new String()`, `new Number()`, `new Boolean()` instances. Their `typeof` also returns `'object'`. Maybe something a bit more strict to only allow simple arrays and objects would be better?

```
if (['Array', 'Object'].includes(value.constructor.name)) {
```

Reply:
I don't think these cases should be an issue because the body is transferred as JSON and so you get:
```
JSON.parse(JSON.stringify({
    s: new String('x'),
    n: new Number(1),
    b: new Boolean(true),
    d: new Date(),
}))
// {s: 'x', n: 1, b: true, d: '2023-04-26T11:48:47.461Z'}
```
where the object versions of primitives are not preserved

Reply:
`new String()` `new Number()` and  `new Boolean()` are rejected by our linter. Anyway even passing them as Object will stringify them fine.